File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.cpp
class Solution {
public:
    int maxProfit(vector<int>& nums) {
        int ans = 0;
        int curr_low=nums[0];   
        for(int i=1;i<nums.size();++i){
            ans = max(ans, nums[i]-curr_low);
            curr_low= min(curr_low,nums[i]);
        }
        return ans;
    }
};

File: Data Structures & Algorithms/maximum-subarray/submission-1.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        ans, temp_ans = -1e10, 0
        for i in range(len(nums)):
            temp_ans += nums[i]
            ans = max(ans, temp_ans)
            temp_ans = max(temp_ans,0)
        return ans